db/migrate/20160517154102_add_twilio_number_to_location.rb

```diff
@@ -0,0 +1,53 @@
+require 'yaml'
+
+class AddTwilioNumberToLocation < ActiveRecord::Migration
+  def up
+    add_column :locations, :twilio_number, :string
+    add_column :locations, :extension, :string
+
+    Location.reset_column_information
+
+    ActiveRecord::Base.transaction do
+      # set the twilio numbers
+      redirects = YAML.load(File.read(Rails.root.join('db/migrate/redirects.yaml')))
+      twilio_lookup = redirects.each_with_object({}) do |r, h|
+        h[r['id']] = r
+      end
+
+      Location.current.each do |location|
+        changes = {}
+        if location.phone =~ /ext/
+          phone, ext = *location.phone.split('ext')
+          changes['extension'] = ext.strip
+          changes['phone'] = phone.strip
+        end
+        if location.phone.present? && location.phone !~ /\A\+44/
+          changes['phone'] = (changes['phone'] || location.phone).delete(' ').gsub(/\A0/, '+44')
+        end
+        if twilio_number = twilio_lookup.delete(location.uid)
+          changes['twilio_number'] = '+' + twilio_number['twilio'].to_s
+        else
+          location.hidden || puts("No twilio number for: #{location.title}")
+        end
+
+        updater = UpdateLocation.new(location: location, user: User.find_by(email: 'david.henry@pensionwise.gov.uk'))
+        if (updater.update(changes) == location) && changes.any?
```

Comment:
modified condition order to make this less relevant. 

As discussed this would not be used in production code, but refactoring for migration is not required at this stage.